<launch>

    <arg name="settings" default="r200_rgbd"/>

    <arg name="map_frame" default="map"/>
    <arg name="target_frame" default="base_footprint"/>
    <arg name="camera_frame" default="camera_link"/>
    <arg name="load" default="false"/>

    <param name="vocab_path" type="string" value="/home/lmh/ORB_SLAM3/Vocabulary/ORBvoc.txt"/>
    <param name="settings_path" type="string" value="/home/lmh/ORB_SLAM3/Examples_old/RGB-D/$(arg settings).yaml"/>
    <param name="map_frame_id" type="string" value="$(arg map_frame)"/>
    <param name="camera_frame_id" type="string" value="$(arg camera_frame)"/>
    <param name="target_frame_id" type="string" value="$(arg target_frame)"/>
    <param name="use_viewer" type="boolean" value="false"/>
    <param name="use_mapper" type="boolean" value="true"/>
    <param name="publish_pointcloud" type="boolean" value="true"/>
    <param name="publish_image" type="boolean" value="true"/>
    <param name="transform_tolerance " type="double" value="0.1"/>

    <node pkg="ORB_SLAM3" name="RGBD" type="RGBD2" output="screen" launch-prefix="gdb -ex run --args">
        <remap from="/camera/depth_registered/image_raw" to="/camera/depth/image_raw"/>
        <remap from="/camera/rgb/image_raw" to="/camera/rgb/image_raw"/>
    </node>

</launch>
